examples/minimalistic_solver/minimalistic_solver.cpp

```diff
@@ -0,0 +1,112 @@
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+/*****************************<COMPILATION>***********************************
+The easiest way to build the example solver is to use the script provided:
+./build.sh <PATH_TO_GINKGO_BUILD_DIR>
+
+Ginkgo should be compiled with `-DBUILD_REFERENCE=on` option.
+
+Alternatively, you can setup the configuration manually:
+
+Go to the <PATH_TO_GINKGO_BUILD_DIR> directory and copy the shared
+libraries located in the following subdirectories:
+
+    + core/
+    + core/device_hooks/
+    + reference/
+    + omp/
+    + cuda/
+
+to this directory.
+
+Then compile the file with the following command line:
+
+c++ -std=c++11 -o minimalistic_solver minimalistic_solver.cpp -I../.. \
+    -L. -lginkgo -lginkgo_reference -lginkgo_omp -lginkgo_cuda
+
+(if ginkgo was built in debug mode, append 'd' to every library name)
+
+Now you should be able to run the program using:
+
+env LD_LIBRARY_PATH=.:${LD_LIBRARY_PATH} ./minimalistic_solver
+
+*****************************<COMPILATION>**********************************/
+
+/*****************************<DECSRIPTION>***********************************
+This is a minimal example that solves a system with Ginkgo. The matrix, right
+hand side and initial guess are read from standard input, and the result is
+written to standard output. The system matrix is stored in CSR format, and the
+system solved using the CG method, preconditioned with the Block-Jacobi
+preconditioner. All computations are done on the GPU.
+
+The easiest way to use the example data from the data/ folder is to concatenate
+the matrix, the right hand side, and the initial solution (in that exact order),
+and pipe the result to the minimalistic_solver executable:
+
+cat data/A.mtx data/b.mtx data/x0.mtx | ./minimalistic_solver
+
+*****************************<DECSRIPTION>**********************************/
+
+#include <ginkgo.hpp>
+#include <iostream>
+
+int main()
+{
+    // Instantiate a CUDA executor
+    auto exec = gko::CudaExecutor::create(0, gko::OmpExecutor::create());
+    // Read data
+    auto A = gko::read<gko::matrix::Csr<>>(std::cin, exec);
+    auto b = gko::read<gko::matrix::Dense<>>(std::cin, exec);
+    auto x = gko::read<gko::matrix::Dense<>>(std::cin, exec);
+    // Create the solver
+    auto solver =
+        gko::solver::Cg<>::Factory::create()
+            .with_preconditioner(
+                gko::preconditioner::BlockJacobiFactory<>::create(exec, 32))
+            .with_criterion(
+                gko::stop::Combined::Factory::create()
+                    .with_criteria(
+                        gko::stop::Iteration::Factory::create()
+                            .with_max_iters(20u)
```

Comment:
I did not notice before we write "(20u)" for 20 iterations. what is the u for?

Reply:
The `u` is for unsigned. Due to `max_iters` being `size_type` which is typically some unsigned number under the hood, you cannot simply type `20`, you have to make sure the number is unsigned, either with static cast or adding the `u` suffix as is done here.

Reply:
thanks! is that documented somewhere?

Reply:
Well the compiler gives you an error and all examples and test use this. But I don't think I added a comment specifically about that anywhere. I can add it.